kratos/utilities/function_parser_utility.cpp

```diff
@@ -23,6 +23,30 @@
 namespace Kratos
 {
 
+// A check only to check that the x is not part of exp
+inline bool CheckThereIsNotx(const std::string& rString)
```

Comment:
as is it seems to only check for the first x

what abt sth like `exp(2)*x`?

=> maybe you could improve the algorithm by removing all occurrences of `exp` and then checking again
You could even implement it in a way that it removes all fct names before checking (and the fct names to be removed could be extended if necessary)

Reply:
My code checks all x, see again, the first line only check if thee is at least one x

Reply:
> i.e.
> 
> ```c++
> for (const auto& r_keyword : {"exp"}) {
>     StringUtilities::ReplaceAllSubstrings(r_keyword, rString, "");
> }
> ```

I don't want to do that, I have bad experience in the past